docs/sdg/sdg-resume.md

```diff
@@ -0,0 +1,146 @@
+# Resuming Interrupted Synthetic Data Generation Runs
+
+|                |            |
+| -------------- | ---------- |
+| Date           | 2024-09-19 |
+| Scope          | Synthetic Data Generation |
+| Status         | Draft |
+| Authors        | [Ben Browning](@bbrowning) |
+
+## What
+
+Synthetic Data Generation (SDG) should track its progress and allow
+users to resume interrupted runs without needing to start over from
+the beginning.
+
+## Why
+
+SDG jobs are long-running, and may occasionally get interrupted. This
+could be because it's running on a cloud spot instance, on a person's
+laptop or machine that needs to shut down for the day, or perhaps a
+GPU hardware failure, out of storage condition, or numerous other
+reasons that could interrupt a run and cause the Python processes or
+entire machine to die.
+
+Because these are long-running jobs that can take hours (or even days,
+in some cases), being able to resume from where we left off is
+important from a usability standpoint. Otherwise, interrupted SDG
+causes a large usability issue and wall clock time for re-work.
+
+## Goals
+
+The user should be able to resume from any long-running point of the
+generation process from where it left off. This includes resuming
+after individual blocks of a pipeline, even if the entire pipeline has
+not finished yet.
+
+The user should have a command like `ilab data generate --resume` to
+explicitly indicate they want to resume an interrupted SDG run. This
+is just an example, and the exact flag/CLI invocation to trigger this
+should be discussed.
+
+The following types of interruption should be handled:
+
+* user-triggered interruptions (e.g. CTRL-c)
+* SIGINT interruptions (e.g. when running on a spot instance of Pod
+  which receives SIGINT)
+* system hardware failure (GPU failure, power loss)
+
+## Non-Goals
+
+This is not replacing checkpoints. Checkpoints are how we save each
```

Comment:
It might be helpful to give the context of how long a batch might take for comparison.

Reply:
Yes, that's a good idea. And I also need to talk about how checkpoints persist across individual data generation runs, while this is focused on the ability to resume a specific interrupted SDG run.